cmd/storage/postgresql/external.go

```diff
@@ -0,0 +1,197 @@
+package storage
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"time"
+
+	add "github.com/Tanya1515/gophermarket/cmd/additional"
+)
+
+func (db *PostgreSQL) RegisterNewUser(user add.User) error {
+
+	_, err := db.dbConn.Exec("INSERT INTO users (login, password, sum, with_drawn) VALUES($1,crypt($2, gen_salt('xdes')),$3,$4)", user.Login, user.Password, 0, 0)
+
+	if err != nil {
+		return fmt.Errorf("error while inserting user with login %s: %w", user.Login, err)
+	}
+	return nil
+}
+
+func (db *PostgreSQL) AddNewOrder(login string, orderNumber string) (err error) {
+
+	var id string
+
+	rows, err := db.dbConn.Query("SELECT orders.id FROM orders JOIN users ON user_id=users.id WHERE users.login=$1", login)
```

Comment:
```suggestion
	err := db.dbConn.QueryRowContext(ctx, "SELECT orders.id AS order_id, users.login FROM orders JOIN users ON orders.user_id=users.id WHERE orders.id=$1", orderNumber).Scan(&order_id, &user_login)
```
- Если просто err - то беда, выходим, но
- Если SqlNoRows, то все хорошо.
- Если строка в БД есть, то есть конфликт. Если логин совпадает, то конфликт по номеру заказа, если не совпадает, то по логину.

Одного запроса на все предварительные проверки достаточно. Дальше можно делать вставку.